state/machine.go

```diff
@@ -2181,6 +2172,21 @@ func (m *Machine) SetUpgradeCharmProfile(appName, chURL string) error {
 	return nil
 }
 
+// SetUpgradeCharmProfileOp returns a transaction for the machine to
+// trigger a change to its LXD Profile(s).
+func (m *Machine) SetUpgradeCharmProfileOp(appName, chURL string) txn.Op {
+	return txn.Op{
+		C:      machinesC,
+		Id:     m.doc.DocID,
+		Assert: bson.D{{"life", Alive}},
+		Update: bson.D{{"$set", bson.D{
+			{"upgradecharmprofilecharmurl", chURL},
+			{"upgradecharmprofileapplication", appName},
+			{"upgradecharmprofilecomplete", ""},
```

Comment:
@SimonRichardson, do you think this will trigger the upgrade charm profile notification watcher prematurely?

Reply:
Don't think so, as the watcher isn't running all the time... It's only run when upgrading